src/lxc/state.c

```diff
@@ -115,13 +115,26 @@ extern int lxc_wait(const char *lxcname, const char *states, int timeout,
 	if (fillwaitedstates(states, s))
 		return -1;
 
-	state = lxc_cmd_sock_get_state(lxcname, lxcpath, s, timeout);
-	if (state < 0) {
-		SYSERROR("failed to receive state from monitor");
-		return -1;
+	for (;;) {
+		state = lxc_cmd_sock_get_state(lxcname, lxcpath, s, timeout);
+		if (state >= 0)
+			break;
+
+		if (errno != ECONNREFUSED) {
+			SYSERROR("Failed to receive state from monitor");
+			return -1;
+		}
+
+		sleep(1);
```

Comment:
Shouldn't the sleep happen after the timeout check?

Reply:
fixed